File: Algorithm_59_Spiral Matrix II.py
class Solution:
    def generateMatrix(self, n):
        if n==0: return []
        if n==1: return[[1]]
        matrix=[[-1 for i in range(n)]for j in range(n)]
        count,cur=1,0
        while cur<n//2:
            for j in range(cur,n-cur):
                matrix[cur][j]=count
                count+=1
            for i in range(cur+1,n-cur):
                matrix[i][n-cur-1]=count
                count+=1
            for j in range(n-cur-2,cur-1,-1):
                matrix[n-cur-1][j]=count
                count+=1
            for i in range(n-cur-2,cur,-1):
                matrix[i][cur]=count
                count+=1
            cur+=1
        if n%2==1: matrix[n//2][n//2]=count
        return matrix

File: Algorithm_59_Spiral Matrix II.java
// 思路：与Algorithm_54采用同样的方法遍历矩阵，变量count记录当前需要放入矩阵的元素

class Solution {
    public int[][] generateMatrix(int n) {
        int[][] result = new int[n][n];
        int topRow = 0, bottomRow = n - topRow - 1, leftCol = topRow, rightCol = n - topRow - 1;
        int count = 1;
        while (topRow <= bottomRow && leftCol <= rightCol) {
            for (int col = leftCol; col <= rightCol; col++) {
                result[topRow][col] = count++;
            }
            for (int row = topRow + 1; row <= bottomRow; row++) {
                result[row][rightCol] = count++;
            }
            if (bottomRow > topRow) {
                for (int col = rightCol - 1; col >= leftCol; col--) {
                    result[bottomRow][col] = count++;
                }
            }
            if (leftCol < rightCol) {
                for (int row = bottomRow - 1; row >= topRow + 1; row--) {
                    result[row][leftCol] = count++;
                }
            }
            topRow++; bottomRow--; leftCol++; rightCol--;
        }
        return result;
    }
}